sympy/printing/pretty/pretty.py

```diff
@@ -1933,8 +1933,7 @@ def _print_Fraction(self, expr):
 
     def _print_ProductSet(self, p):
         if len(p.sets) >= 1 and not has_variety(p.sets):
-            from sympy import Pow
-            return self._print(Pow(p.sets[0], len(p.sets), evaluate=False))
+            return self._print(p.sets[0]) ** self._print(len(p.sets))
```

Comment:
Why did this need to be changed?

Reply:
We can't use `p.sets[0]` which is a `Set` in the args of a `Pow`.

Reply:
Oh..I get it. It is now a Set "Pow"  not and Expr Pow.